### DISCUSSION TITLE
Time plan for a new release

### DISCUSSION THREAD

**Discussion by madanyang** [plausible/analytics#3793] [Self-Hosted Support]
v2 was relased back in 07-2023. Is there a time plan for a new release. Like excluding self ip in settings would be nice to have

**Comment by ruslandoga** [comment]
👋 @madanyang 

It's in the works :)

**Reply by ruslandoga** [reply]
If all goes well, ETA for v2.1.0 is later next week.

**Reply by raabmar** [reply]
@ruslandoga that's good news. I am following this discussion https://github.com/plausible/analytics/discussions/3691
 and it confused me a lot with all the different options to update. None of it seemed like the official recommended way to update, but more like a way around it to somehow pull it together. 

I know there are just 3 lines in the console for the Docker image to update self-hosted Plausible. I have those from other discussion in here, so if someone else reads it, don't rely on them
```
docker-compose down --remove-orphans
docker-compose pull plausible
docker-compose up -d
```

Will I be able to update to the latest version v2.1.0 this way?

**Reply by ruslandoga** [reply]
[First v2.1.0 release candidate](https://github.com/plausible/analytics/discussions/3817) is now available :)

**Reply by ruslandoga** [reply]
Yes, you should be able to update like this once you change the image for the `plausible` service.

**Reply by raabmar** [reply]
what's the difference then between those two
```
docker-compose down --remove-orphans
docker-compose pull plausible
docker-compose up -d
```
and from the official instruction for updating to v2.1.0-rc.0 https://github.com/plausible/analytics/discussions/3817
```
docker compose stop plausible plausible_events_db
docker compose rm plausible plausible_events_db
docker compose up -d
```

**Reply by ruslandoga** [reply]
Since I'm not very good with words I asked your question to ChatGPT and it came up with this:

---

### Key Differences
- Granularity: The first sequence is more of a blanket approach, stopping and removing all containers managed by Docker Compose, including orphan containers. The second sequence is more targeted, only stopping and removing specific services. This can be useful if you want to update certain parts of your application without affecting others.
- Orphan Containers: The first sequence explicitly removes orphan containers, which might not be directly related to the services you're updating. The second sequence does not, as it focuses on specific containers.
- Pulling Images: The first sequence includes a step to explicitly pull a new image for the plausible service. The second sequence relies on Docker Compose's behavior to automatically use the latest image when recreating containers, assuming the image is specified in the docker-compose.yml or an image update isn't required as per the instructions.

In summary, the choice between these sequences depends on your specific needs for updating or managing your Docker environment. The first is more comprehensive and might be suitable for broader updates or cleanup, while the second offers more precision, potentially reducing downtime for unaffected services.

---

I agree with it.

**Reply by raabmar** [reply]
OK, I asked because there are basically two different instructions, both confirmed that those are the way to do it. Both of those set of commands are from within the github discussions, both confirmed that they should work to update (once on version 2.0.x). For me it was just guessing to pick which one could work. I'll try another time to update - if anyone else succeeds, perhaps post and confirm in here please, or share what issues you run into. Thanks

**Reply by ruslandoga** [reply]
Both of these commands lead to the same end result.
